csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -253,14 +253,25 @@ public TBuilder Resize(int length)
 
             public TBuilder Swap(int i, int j)
             {
-                // TODO: Implement
-                throw new NotImplementedException();
+                byte ti = ValueBuffer.Span[i];
+                byte tj = ValueBuffer.Span[j];
+                ValueBuffer.Span[i] = tj;
+                ValueBuffer.Span[j] = ti;
+                return Instance;
             }
 
             public TBuilder Set(int index, byte value)
             {
-                // TODO: Implement
-                throw new NotImplementedException();
+                ValueBuffer.Span[index] = value;
+                ValidityBuffer.Set(index);
+                return Instance;
+            }
+
+            public TBuilder SetNull(int index)
+            {
+                ValueBuffer.Span[index] = default;
+                ValidityBuffer.Set(index, false);
```

Comment:
The indexes in the validity buffer don't align with the indexes in the value buffer unless each element of the binary array has a length of exactly one.

To implement SetNull correctly, if the previous value was non-null then it would be necessary to compact the value buffer and adjust all of the offsets in ValueOffsets.

Reply:
(The same comment roughly applies to the implementation of Set, above.)